src/encoded/static/components/genetic_modification.js

```diff
@@ -316,190 +316,229 @@ Attribution.propTypes = {
-            ]}
-        />
-    );
+        return (
+            <DocumentsPanel
+                documentSpecs={[
+                    { label: 'Characterization', documents: characterizationsCopy },
+                    { label: 'Modification document', documents: modificationDocuments },
+                ]}
+            />
+        );
+    }
+    return null;
 };
 
 DocumentsRenderer.propTypes = {
-    modCharDocs: PropTypes.object, // GM document search results containing GM docs and GM characterization docs
     characterizations: PropTypes.array.isRequired, // GM characterizations
-    charDocs: PropTypes.array.isRequired, // Array of @ids of docs belonging to all GM characterizations
-};
-
-DocumentsRenderer.defaultProps = {
-    modCharDocs: null,
+    characterizationDocuments: PropTypes.array.isRequired, // GM characterization documents for all given characterizations
+    modificationDocuments: PropTypes.array.isRequired, // GM documents
 };
 
 
 // Render the entire GeneticModification page. This is called by the back end as a result of an
 // attempt to render an object with an @type of GeneticModification.
-export const GeneticModificationComponent = (props, reactContext) => {
-    const { context } = props;
-    const itemClass = globals.itemClass(context, 'view-detail key-value');
-
-    // Configure breadcrumbs for the page.
-    const crumbs = [
-        { id: 'Genetic Modifications' },
-        { id: context.target && context.target.label, query: `target.label=${context.target && context.target.label}`, tip: context.target && context.target.label },
-        { id: context.modification_type, query: `modification_type=${context.modification_type}`, tip: context.modification_type },
-    ];
-
-    // Collect and combine document @ids, including from genetic modification
-    // characterization document @ids. Later we'll do a GET request on these @ids to show these
-    // documents along with the embedded characterizations.
-    let modDocs = context.documents && context.documents.length ? context.documents : [];
-    let charDocs = [];
-    if (context.characterizations && context.characterizations.length) {
-        context.characterizations.forEach((characterization) => {
-            if (characterization.documents && characterization.documents.length) {
-                charDocs = charDocs.concat(characterization.documents);
+class GeneticModificationComponent extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            modificationDocuments: [], // GM document search results
+            characterizationDocuments: [], // GM characterization document search results
+        };
+        this.requestDocuments = this.requestDocuments.bind(this);
+        this.geneticModificationComponentChanged = this.geneticModificationComponentChanged.bind(this);
+    }
+
+    componentDidMount() {
+        this.requestDocuments();
+    }
+
+    componentDidUpdate() {
+        this.requestDocuments();
```

Comment:
I haven't tried it, but it looks like this would result in 2 requests? A change to props would trigger this, which would fetch the query and update the state, which would trigger this again. No?

Reply:
Yes, absolutely, on initial page load. I saw this as not a significant issue. Should I put more time into avoiding that? As I said, I’m fixing one corner case right now.

Reply:
It seems to me that we only want to reissue the request if the props changed, not if the state changed. I'm not sure I understand the update scenario though -- what would cause the props to change in a way that requires reissuing the request?